cpp/src/groupby/new_groupby.cu

```diff
@@ -244,32 +246,37 @@ gdf_unique_indices(cudf::table const& input_table, gdf_context const& context)
   cudaStreamCreate(&stream);
   auto exec = rmm::exec_policy(stream)->on(stream);
 
-  rmm::device_vector<gdf_index_type> unique_indices(nrows);
+  // Allocating memory for GDF column
+  gdf_column unique_indices {};
+  RMM_TRY(RMM_ALLOC(&unique_indices.data, sizeof(gdf_index_type)*nrows, nullptr));
+  unique_indices.dtype = cudf::gdf_dtype_of<gdf_index_type>();
 
   auto counting_iter = thrust::make_counting_iterator<gdf_size_type>(0);
   auto device_input_table = device_table::create(input_table);
   bool nullable = device_input_table.get()->has_nulls();
   if (nullable){
     auto comp = row_equality_comparator<true>(*device_input_table, true);
     result_end = thrust::unique_copy(exec, counting_iter, counting_iter+nrows,
-                              unique_indices.data().get(),
+                              (gdf_index_type *)unique_indices.data,
                               comp);
   } else {
     auto comp = row_equality_comparator<false>(*device_input_table, true);
     result_end = thrust::unique_copy(exec, counting_iter, counting_iter+nrows,
-                              unique_indices.data().get(),
+                              (gdf_index_type *)unique_indices.data,
                               comp);
   }
 
-  unique_indices.resize(thrust::distance(unique_indices.data().get(), result_end));
+  // size of the GDF column is being resized
+  RMM_TRY(RMM_REALLOC(&unique_indices.data, thrust::distance((gdf_index_type *)unique_indices.data, result_end)*sizeof(gdf_index_type), nullptr));
```

Comment:
I believe that `RMM_REALLOC` is broken and does not preserve the original data.

Reply:
I did not see any issues in the test cases

Reply:
I can leave resizing if its actually broken but limit the address access with the size.

Reply:
See https://github.com/rapidsai/rmm/issues/5